zeebe/scheduler/src/test/java/io/camunda/zeebe/scheduler/testing/TestConcurrencyControl.java

```diff
@@ -44,6 +48,35 @@ public <T> void runOnCompletion(
     future.onComplete(callback);
   }
 
+  @Override
+  public <T> void runOnCompletion(
+      final Collection<ActorFuture<T>> actorFutures, final Consumer<Throwable> callback) {
+    if (actorFutures.isEmpty()) {
+      callback.accept(null);
+      return;
+    }
+
+    final var error = new AtomicReference<Throwable>();
+    final var futuresCompleted = new AtomicInteger(actorFutures.size());
+    final var finalFuture = new TestActorFuture<>();
+    for (final ActorFuture<T> f : actorFutures) {
+      f.onComplete(
+          (r, e) -> {
+            if (e != null) {
+              error.set(e);
+            }
+            if (futuresCompleted.decrementAndGet() == 0) {
+              if (error.get() != null) {
+                finalFuture.completeExceptionally(error.get());
+              } else {
+                finalFuture.complete(null);
+              }
```

Comment:
:upside_down_face: Technically not needed, you could just always complete the future successfully with whatever is in `error`.